MainWindow.xaml.cs

```diff
@@ -446,64 +485,77 @@ private void CancelDumping()
+
+            //TODO: a specific KnownSystem to manage Custom would be better
+            bool isCustom = systemTuple.Item1 == "Custom Input";
+            var selectedSystem = systemTuple.Item2;
+            var selectedDiscType = discTypeTuple != null ? discTypeTuple.Item2 : DiscType.NONE;
+
+            // No system chosen, update status
+            if (selectedSystem == KnownSystem.NONE)
             {
-                cmb_DiscType.SelectedIndex++;
-                tuple = cmb_DiscType.SelectedItem as Tuple<string, KnownSystem?, DiscType?>;
+                lbl_Status.Content = "Please select a valid system";
+                btn_StartStop.IsEnabled = false;
             }
-
-            // If we're on an unsupported type, update the status accordingly
-            switch (tuple.Item3)
+            else
             {
-                case DiscType.NONE:
-                    lbl_Status.Content = "Please select a valid disc type";
-                    btn_StartStop.IsEnabled = false;
-                    break;
-                case DiscType.GameCubeGameDisc:
-                case DiscType.GDROM:
-                    lbl_Status.Content = string.Format("{0} discs are partially supported by DIC", Converters.DiscTypeToString(tuple.Item3));
-                    btn_StartStop.IsEnabled = (_drives.Count > 0 ? true : false);
-                    break;
-                case DiscType.HDDVD:
-                case DiscType.UMD:
-                case DiscType.WiiOpticalDisc:
-                case DiscType.WiiUOpticalDisc:
-                    lbl_Status.Content = string.Format("{0} discs are not currently supported by DIC", Converters.DiscTypeToString(tuple.Item3));
-                    btn_StartStop.IsEnabled = false;
-                    break;
-                case DiscType.DVD5:
-                case DiscType.DVD9:
-                    if (tuple.Item2 == KnownSystem.MicrosoftXBOX360XDG3)
-                    {
-                        lbl_Status.Content = string.Format("{0} discs are not currently supported by DIC", Converters.DiscTypeToString(tuple.Item3));
+                // If we're on an unsupported type, update the status accordingly
+                switch (selectedDiscType)
+                {
+                    case DiscType.NONE:
+                        lbl_Status.Content = "Please select a valid disc type";
                         btn_StartStop.IsEnabled = false;
-                    }
-                    else
-                    {
-                        lbl_Status.Content = string.Format("{0} ready to dump", Converters.DiscTypeToString(tuple.Item3));
+                        break;
+                    case DiscType.GameCubeGameDisc:
+                    case DiscType.GDROM:
+                        lbl_Status.Content = string.Format("{0} discs are partially supported by DIC", Converters.DiscTypeToString(selectedDiscType));
                         btn_StartStop.IsEnabled = (_drives.Count > 0 ? true : false);
```

Comment:
Wouldn't the DiscTypeToString calls return the same as discTypeTuple.Item1? That would reduce the number of method calls here.

Reply:
Yes, that's a side effect of the refactor of the top part for the new management, just haven't modified it, but this `discTypeTuple.Item1` is so unreadable, damn unnamed tuples.